### DISCUSSION TITLE
Connecting Geoserver with deck.gl

### DISCUSSION THREAD

**Discussion by sheecegardezi** [visgl/deck.gl#7636] [General]
Geoserver can be connected with deckgl using a tile layer, a basic example using tile layer:

```
        const devicePixelRatio = (typeof window !== "undefined" && window.devicePixelRatio) || 1;

        new TileLayer({
            id: "geoserver-layer",
            data: [
                "http://xyz:8080/geoserver/gwc/service/tms/1.0.0/prod:file@EPSG:4326@png/{z}/{x}/{y}.png"
            ],
            maxRequests: 1000,
            pickable: false,
            tileSize: 512 / devicePixelRatio,
            visible: true,
            renderSubLayers: (props) => {
                const {
                    bbox: { west, south, east, north },
                } = props.tile;
                return [
                    new BitmapLayer(props, {
                        data: null,
                        image: props.data,
                        bounds: [west, south, east, north],
                    }),
                ];
            }
        }),
```

I'm unsure how to get the right size! I just try a few different combinations of multiples like 128, 256, 512.. until it displays correctly.

**Comment by Pueablo** [comment]
> Geoserver se puede conectar con deckgl usando una capa de teselas, un ejemplo básico usando una capa de teselas:
> 
> ```
>         const devicePixelRatio = (typeof window !== "undefined" && window.devicePixelRatio) || 1;
> 
>         new TileLayer({
>             id: "geoserver-layer",
>             data: [
>                 "http://xyz:8080/geoserver/gwc/service/tms/1.0.0/prod:file@EPSG:4326@png/{z}/{x}/{y}.png"
>             ],
>             maxRequests: 1000,
>             pickable: false,
>             tileSize: 512 / devicePixelRatio,
>             visible: true,
>             renderSubLayers: (props) => {
>                 const {
>                     bbox: { west, south, east, north },
>                 } = props.tile;
>                 return [
>                     new BitmapLayer(props, {
>                         data: null,
>                         image: props.data,
>                         bounds: [west, south, east, north],
>                     }),
>                 ];
>             }
>         }),
> ```
> 
> ¡No estoy seguro de cómo obtener el tamaño correcto! Solo pruebo algunas combinaciones diferentes de múltiplos como 128, 256, 512... hasta que se muestra correctamente.

@sheecegardezi  hi, i have a question, how do you get geoserver xyz?

**Comment by sheecegardezi** [comment]
@Pueablo  Geoserver is an application that serves geospatial datasets, similar to how SQL Server serves tables of data. The official documentation for Geoserver, which includes tutorials for basic usage, can be found at https://docs.geoserver.org/. Another helpful resource is the guide available at https://www.e-education.psu.edu/geog585/node/686.

To set up a Geoserver instance locally, you will need to follow these steps:

1. Install Java: Geoserver is a Java-based application, so you'll need to install Java on your system first if it's not already installed.
2. Download Geoserver: You can download Geoserver from the official website (https://geoserver.org/download/) and choose the appropriate version for your operating system.
3. Install Geoserver: Once you have downloaded Geoserver, you can install it on your system by following the installation instructions for your operating system. This usually involves running an installer or extracting the files to a directory.
4. Start Geoserver: After installing Geoserver, you can start it up by running the appropriate command for your operating system. This will launch Geoserver and allow you to access its web interface.
5. Configure Geoserver: Once Geoserver is running, you can configure it by setting up data stores, creating layers, and configuring styles for your data. You can do this using the web interface or by editing configuration files directly.
6. Serve Geospatial Data: Finally, you can serve your geospatial data using Geoserver by publishing layers to the web. This will allow others to view and interact with your data using various web mapping tools.
